app/src/main/java/in/ac/iitb/gymkhana/iitbapp/fragment/MyEventsFragment.java

```diff
@@ -44,7 +71,85 @@ public void onClick(View v) {
             }
         });
         return view;
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        session = new SessionManager(getContext());
+        currentuser = User.fromString(session.pref.getString(Constants.CURRENT_USER, "Error"));
+        appDatabase = AppDatabase.getAppDatabase(getContext());
+        new showEvents().execute();
 
+        feedSwipeRefreshLayout = getActivity().findViewById(R.id.feed_swipe_refresh_layout);
+        feedSwipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                update();
+                feedSwipeRefreshLayout.setRefreshing(false);
+            }
+        });
     }
 
-}
+    private void update(){
```

Comment:
What is this method trying to do exactly? I see the same code in showEvents.doInBackground as well, but none of these make sense to me